src/catchup/CatchupManagerImpl.cpp

```diff
@@ -408,20 +466,21 @@ CatchupManagerImpl::trimSyncingLedgers()
         // This erases [begin, it).
         mSyncingLedgers.erase(mSyncingLedgers.begin(), it);
     };
-    removeLedgersLessThan(mApp.getLedgerManager().getLastClosedLedgerNum() + 1);
-    auto& hm = mApp.getHistoryManager();
+    removeLedgersLessThan(*mLastQueuedToApply + 1);
```

Comment:
Yeah, this is intended (although a bit awkward) - CM bufferes new ledgers sequentially until it reaches a checkpoint boundary. Once we buffer a new checkpoint, we trim everything before that. The idea is that we'll just do catchup again and replay the same ledgers from history. I think the reason this was added is to avoid buffering ledgers in an unbounded manner. Testing on my local validator, the trim-then-catchup mechanism continues to work as expected.